app/jobs/bulkrax/delete_job.rb

```diff
@@ -5,8 +5,16 @@ class DeleteJob < ApplicationJob
     queue_as :import
 
     def perform(entry, importer_run)
-      obj = entry.factory.find
-      obj&.delete
+      obj = entry.factory.class.find(entry.raw_metadata["id"])
```

Comment:
What about writing: `entry.factory.delete`

And then each factory would implement that according to the strategy.